File: src/Application/Clubs/DTOs/UserClubs.cs
using CocktailsApp.Application.Common;


namespace CocktailsApp.Application.Clubs
{
    public sealed record UserClubs(
        Guid UserId,
        IReadOnlyCollection<UserClubItem> Clubs
    ) : IDTO;

    public sealed record UserClubItem(
        Guid ClubId,
        string ClubName,
        string ClubDescription,
        bool IsOwner,
        string? ImageId,
        IReadOnlyCollection<string> RoleNames,
        IReadOnlyCollection<string> EffectivePermissions
    ) : IDTO;
}

File: client/src/app/features/user/models/user-clubs.ts
export interface UserClubs {
  userId: string;
  clubs: UserClubItem[];
}

export interface UserClubItem {
  clubId: string;
  clubName: string;
  clubDescription: string;
  isOwner: boolean;
  imageUrl:string;
  roleNames: string[];
  effectivePermissions: string[];
}
